File: Generating/prototype/python/circle.py
from shape import Shape
from math import pow, pi

class Circle(Shape):
  radius = None
  
  def __init__(self, params = None):
    if params is not None:
      self.radius = params['radius']
    pass
  
  def square(self):
    return pi * pow(self.radius, 2)

  def clone(self):
    return Circle({
      'x': self.x,
      'y': self.y,
      'color': self.color,
      'radius': self.radius
    })

  def equals(self, targetObject):
    if targetObject.radius != self.radius:
      return False
    if targetObject.x != self.x:
      return False
    if targetObject.y != self.y:
      return False
    if targetObject.color != self.color:
      return False
    
    return True

File: Generating/prototype/src/Circle.ts
import Shape from './Shape'
import { CircleParams, ICircle } from './interfaces'

export default class Circle extends Shape implements ICircle {
  private radius: number
  
  constructor(params: CircleParams) {
    super(params)
    this.radius = (!params.radius) ? 0 : params.radius
  }

  getRadius(): number { return this.radius }
  
  clone(): ICircle { return Object.create(this) }
  square(): number { return Math.PI * Math.pow(this.radius, 2) }
  deepEquals(circleInstance: ICircle): boolean {
    switch (true) {
      case (circleInstance.getX() !== this.x):
        return false
      case (circleInstance.getY() !== this.y):
        return false
      case (circleInstance.square() !== this.square()):
        return false
      case (circleInstance.getColor() !== this.color):
        return false
      default:
        return true
    }
  }
}